File: app/src/main/java/dev/ragnarok/fenrir/fragment/search/criteria/NewsFeedCriteria.java
package dev.ragnarok.fenrir.fragment.search.criteria;

import android.os.Parcel;

import dev.ragnarok.fenrir.R;
import dev.ragnarok.fenrir.fragment.search.options.SimpleDateOption;
import dev.ragnarok.fenrir.fragment.search.options.SimpleGPSOption;

public final class NewsFeedCriteria extends BaseSearchCriteria {

    public static final int KEY_GPS = 1;
    public static final int KEY_START_TIME = 2;
    public static final int KEY_END_TIME = 3;

    public static final Creator<NewsFeedCriteria> CREATOR = new Creator<NewsFeedCriteria>() {
        @Override
        public NewsFeedCriteria createFromParcel(Parcel in) {
            return new NewsFeedCriteria(in);
        }

        @Override
        public NewsFeedCriteria[] newArray(int size) {
            return new NewsFeedCriteria[size];
        }
    };

    public NewsFeedCriteria(String query) {
        super(query);

        appendOption(new SimpleGPSOption(KEY_GPS, R.string.gps, true));
        appendOption(new SimpleDateOption(KEY_START_TIME, R.string.date_start, true));
        appendOption(new SimpleDateOption(KEY_END_TIME, R.string.date_to, true));
    }

    private NewsFeedCriteria(Parcel in) {
        super(in);
    }

    @Override
    public int describeContents() {
        return 0;
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/fragment/search/criteria/NewsFeedCriteria.kt
package dev.ragnarok.fenrir.fragment.search.criteria

import android.os.Parcel
import android.os.Parcelable
import dev.ragnarok.fenrir.R
import dev.ragnarok.fenrir.fragment.search.options.SimpleDateOption
import dev.ragnarok.fenrir.fragment.search.options.SimpleGPSOption

class NewsFeedCriteria : BaseSearchCriteria {
    constructor(query: String?) : super(query) {
        appendOption(SimpleGPSOption(KEY_GPS, R.string.gps, true))
        appendOption(SimpleDateOption(KEY_START_TIME, R.string.date_start, true))
        appendOption(SimpleDateOption(KEY_END_TIME, R.string.date_to, true))
    }

    internal constructor(parcel: Parcel) : super(parcel)

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        const val KEY_GPS = 1
        const val KEY_START_TIME = 2
        const val KEY_END_TIME = 3

        @JvmField
        val CREATOR: Parcelable.Creator<NewsFeedCriteria> =
            object : Parcelable.Creator<NewsFeedCriteria> {
                override fun createFromParcel(parcel: Parcel): NewsFeedCriteria {
                    return NewsFeedCriteria(parcel)
                }

                override fun newArray(size: Int): Array<NewsFeedCriteria?> {
                    return arrayOfNulls(size)
                }
            }
    }
}